monty/shutil.py

```diff
@@ -75,15 +77,28 @@ def compress_file(
         filepath (str | Path): Path to file.
         compression (str): A compression mode. Valid options are "gz" or
             "bz2". Defaults to "gz".
+        target_dir (str | Path): An optional target dir where the result compressed
+            file would be stored. Defaults to in-place compression.
     """
     filepath = Path(filepath)
-    if compression not in ["gz", "bz2"]:
+    target_dir = str(target_dir)
```

Comment:
Most of the operations in the module are using the `pathlib` library, so it seems better to convert this to `Path` and use such operations throughout.

Reply:
Yes I find this funny too... But I need to rethink about how to handle and detect the default value if using `Path`

Reply:
This should be self explanatory with a `None` default.